### PR TITLE
add heading angle of slaves to metadata

### PR DESCRIPTION
**Pull request by estebanag**
This is required for https://senbox.atlassian.net/browse/SITBX-730

### CODE REVIEW

File: snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java [hunk 1]
```diff
@@ -415,6 +416,17 @@ public void initialize() throws OperatorException {
         }
 
         setAutoGrouping();
+
+        // Add heading angles
+        final MetadataElement slavesHeadingAnglesElem = new MetadataElement("Slaves_Heading_Angles");
+        final MetadataElement abstractedMetadataTarget = targetProduct.getMetadataRoot().getElement("Abstracted_Metadata");
```
Comment (lveci) [inline snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java:422]: @estebanag collocate is primarily an optical operator and optical products may not have an Abstracted_Metadata element. This needs to check if the element exists

Comment (marpet) [reply snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java:422]: Yes, it should be checked if targetProduct and slaveProducts have Abstracted_Metadata and only then the heading elements should be added. If this is done then this is okay for me.

**Review by marpet — CHANGES_REQUESTED** [review]
See comment

File: snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java (hunk 1 above)
Comment (estebanag) [reply snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java:421]: I just updated it. Is this what you had in mind?

Comment (marpet) [reply snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java:421]: Looks good. Have you tested with optical products?

**Review by marpet — APPROVED** [review]
When tested with optical data then this change can be merged.

**Review by lveci — APPROVED** [review]

File: snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java (hunk 1 above)
Comment (estebanag) [reply snap-collocation/src/main/java/org/esa/snap/collocation/CollocateOp.java:421]: Just ran it with Sentinel-2 data and it went well. I also debugged it, and the check for `Abstracted_Metadata` works as expected.

**Pull request merged**